js/atoms/src/SelectSource.svelte

```diff
@@ -1,22 +1,30 @@
 <script lang="ts">
-	import { Microphone, Upload, Video } from "@gradio/icons";
+	import { Microphone, Upload, Webcam, ImagePaste } from "@gradio/icons";
 
-	export let sources: string[];
-	export let active_source: string;
+	type sources = "upload" | "microphone" | "webcam" | "clipboard" | null;
+
+	export let sources: Partial<sources>[];
+	export let active_source: Partial<sources>;
 	export let handle_clear: () => void = () => {};
+	export let handle_select: (source_type: Partial<sources>) => void = () => {};
+
+	let unique_sources = [...new Set(sources)];
```

Comment:
not reactive, right?
```suggestion
	$: unique_sources = [...new Set(sources)];
```
(not completely sure if this deduplication logic is even necessary tbh? no particular strong feelnig about it tho)